src/prism.c

```diff
@@ -22161,6 +22098,10 @@ pm_parser_init(pm_parser_t *parser, const uint8_t *source, size_t size, const pm
         .warn_mismatched_indentation = true
     };
 
+    // TODO: find a better starting size
```

Comment:
👍 

4k is probably a fine starting size, maybe we can add some logging later and figure out a better start.